### ISSUE TITLE
False negative for non-overlapping type check inside function with strict_equality flag

### ISSUE THREAD

**Issue by Garrett-R** [python/mypy#10055]
**Bug Report**

**To Reproduce**

Run this file to `test.py`:

```python
if 'red' == 5:  # <-- Mypy finds error
    print('equal')

def foo():
    if 'red' == 5:  # <-- Mypy does not find error
        print('equal')

foo()
```

then execute

```shell
mypy test.py --strict-equality
```
**Expected Behavior**

Mypy should find 2 errors in this file.

**Actual Behavior**

Mypy fails to find the error inside the function.

**Your Environment**

- Mypy version used: 0.8000
- Mypy command-line flags: `--strict-equality`
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.9.1
- Operating system and version: Ubuntu 18.04

**Comment by hauntsaninja** [comment]
First item of https://mypy.readthedocs.io/en/stable/common_issues.html#no-errors-reported-for-obviously-wrong-code

**Comment by Garrett-R** [comment]
Ah, my apologies!  Thank you for your time and thanks for the package!
